src/util/SpawnPed.cpp

```diff
@@ -1,48 +1,71 @@
-#include "core/frontend/Notifications.hpp"
-#include "game/backend/FiberPool.hpp"
-#include "game/backend/ScriptMgr.hpp"
-#include "game/rdr/Natives.hpp"
-
+#include "SpawnPed.hpp"
 
 namespace YimMenu
 {
-	bool SpawnPed(std::string model_name, Player player, int playerPed)
+	int SpawnPed(std::string model_name, Vector3 coords, float heading, bool blockNewPedMovement, bool spawnDead, bool invincible, bool invisible, int scale)
 	{
-		FiberPool::Push([=] {
-			Hash model = MISC::GET_HASH_KEY(model_name.c_str());
-			if (STREAMING::IS_MODEL_IN_CDIMAGE(model) && STREAMING::IS_MODEL_VALID(model))
+		Hash model = MISC::GET_HASH_KEY(model_name.c_str());
+		if (STREAMING::IS_MODEL_IN_CDIMAGE(model) && STREAMING::IS_MODEL_VALID(model))
+		{
+			for (int i = 0; i < 5 && !STREAMING::HAS_MODEL_LOADED(model); i++)
+			{
+				STREAMING::REQUEST_MODEL(model, false);
+				ScriptMgr::Yield();
+			}
+
+			auto ped = PED::CREATE_PED(model, coords.x, coords.y, coords.z, heading, 1, 0, 0, 0);
+
+			ScriptMgr::Yield();
+			PED::_SET_RANDOM_OUTFIT_VARIATION(ped, true);
+			ENTITY::SET_ENTITY_ALPHA(ped, 255, 0);
+			ENTITY::PLACE_ENTITY_ON_GROUND_PROPERLY(ped, true);
+			ScriptMgr::Yield();
+			NETWORK::NETWORK_REGISTER_ENTITY_AS_NETWORKED(ped);
+
+			int id = NETWORK::PED_TO_NET(ped);
+			if (NETWORK::NETWORK_DOES_NETWORK_ID_EXIST(id))
 			{
-				if (!STREAMING::HAS_MODEL_LOADED(model))
+				ENTITY::SET_ENTITY_SHOULD_FREEZE_WAITING_ON_COLLISION(ped, true);
+				if (NETWORK::NETWORK_GET_ENTITY_IS_NETWORKED(ped))
 				{
-					STREAMING::REQUEST_MODEL(model, false);
-					ScriptMgr::Yield();
+					NETWORK::SET_NETWORK_ID_EXISTS_ON_ALL_MACHINES(id, true);
+					NETWORK::IS_NETWORK_ID_OWNED_BY_PARTICIPANT(id);
 				}
+			}
+			ScriptMgr::Yield();
 
-				Vector3 coords       = ENTITY::GET_OFFSET_FROM_ENTITY_IN_WORLD_COORDS(playerPed, 0.0, 3.0, -0.3);
-				float playerRotation = ENTITY::GET_ENTITY_ROTATION(playerPed, 2).x;
-				auto ped             = PED::CREATE_PED(model, coords.x, coords.y, coords.z, playerRotation, 0, 0, 0, 0);
+			if (blockNewPedMovement == true)
+			{
+				ENTITY::FREEZE_ENTITY_POSITION(ped, true);
+			}
 
-				ScriptMgr::Yield();
-				PED::_SET_RANDOM_OUTFIT_VARIATION(ped, true);
-				ENTITY::SET_ENTITY_ALPHA(ped, 255, 0);
+			if (spawnDead == true)
```

Comment:
These values can be used within the natives. No need to scope each factor.

Reply:
What do you mean? Sorry Im a little confused.

Reply:
Instead of an if check, the value can be used in the parameters of the native. Such as ENTITY::SET_ENTITY_INVINCIBLE(ped, **invincible**)

Reply:
Okay, thanks for the explanation. I will have to keep the check for damaging the ped though I believe.